app/partials/coinify-modal.pug

```diff
@@ -1,106 +1,60 @@
-    button.button-muted(ng-show="onStep('summary') && !error" ng-click="goTo('select-payment-medium')" translate="GO_BACK")
-    .flex-1.flex-end(ng-hide="status.loading")
-      button.button-muted.mrm(ng-click="$close() && goToOrderHistory()" translate="CLOSE" ng-show="onStep('trade-formatted') && bitcoinReceived")
-      button.button-muted.mrm(ng-click="close()" translate="CANCEL_ORDER" ng-hide="afterStep('summary') || onStep('summary') && error")
-      button.button-muted.mrm(ng-click="close() && goToOrderHistory()" translate="FINISH_LATER" ng-show="onStep('isx')")
-      div(ng-hide="afterStep('summary')")
-        button.button-primary(
-          ladda-translate="CONTINUE"
-          ng-click="nextStep()"
-          data-style="expand-left"
-          ui-ladda="status.waiting"
-          ng-disabled="isDisabled()"
-          ng-show="onStep('select-country')")
-        button.button-primary(
-          ladda-translate="CONTINUE"
-          ng-click="nextStep()"
-          data-style="expand-left"
-          ui-ladda="status.waiting"
-          ng-disabled="isDisabled()"
-          ng-show="onStep('email')")
-        button.button-primary(
-          ladda-translate="CONTINUE"
-          ng-click="signup()"
-          data-style="expand-left"
-          ui-ladda="status.waiting"
-          ng-disabled="isDisabled()"
-          ng-show="onStep('accept-terms')")
-        button.button-primary(
-          ladda-translate="CONTINUE"
-          ng-click="confirmOrContinue()"
-          data-style="expand-left"
-          ui-ladda="status.waiting"
-          ng-disabled="isDisabled()"
-          ng-show="onStep('select-payment-medium')")
-        button.button-primary(
-          ladda-translate="CONFIRM"
-          ng-click="buy()"
-          data-style="expand-left"
-          ui-ladda="status.waiting"
-          ng-disabled="isDisabled() || fiatFormInvalid || status.gettingQuote"
-          ng-show="onStep('summary') && !error")
-      div(ng-show="onStep('trade-in-review') || onStep('summary') && error || onStep('trade-formatted') && !bitcoinReceived")
-        button.button-primary(translate="CLOSE" ng-click="cancel() && goToOrderHistory()")
-      div(ng-show="onStep('trade-formatted') && bitcoinReceived")
-        button.button-primary(translate="SEE_BITCOIN" ng-hide="inMobileBuy" ng-click="cancel()" ui-sref="wallet.common.transactions")
-        button.button-primary(translate="SEE_BITCOIN" ng-show="inMobileBuy" ng-click="exitToNativeTx()")
+  
+  div(ng-if="onStep('email')" ng-controller="CoinifyEmailController" ng-include="'partials/coinify/email.pug'")
+  div(ng-if="onStep('accept-terms')" ng-controller="CoinifySignupController" ng-include="'partials/coinify/signup.pug'")
+  div(ng-if="onStep('select-payment-medium')" ng-controller="CoinifyMediumController" ng-include="'partials/coinify/medium.pug'")
+  div(ng-if="onStep('summary')" ng-controller="CoinifySummaryController" ng-include="'partials/coinify/summary.pug'")
```

Comment:
Passing state from attrs / callbacks is good because it creates a very clear boundary between the component and the parent controller (although `vm` is usually fine too, as long as children treat it as read-only like we try to do with sfox).

Would be cool to use ui-router. There are some issues I can see with that though. If we start opening the modal with a route change instead of the `modal` service, can we still pass in data like a `trade` object? Would that be done through state params? Also, would we lose the benefit of receiving a promise from opening the modal?